src/test/java/com/game/Battleship/controllers/ShipControllerTests.java

```diff
@@ -0,0 +1,37 @@
+package com.game.Battleship.controllers;
+import com.game.Battleship.models.Ship;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.util.List;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+
+public class ShipControllerTests {
+    @Autowired
+    private TestRestTemplate template;
+
+    @Test
+    public void shipEndpointContains5ShipObjects(){
+
+        List<Object> response = template.getForObject("/ships", List.class);
+        Assertions.assertEquals(5, response.size());
+
+    }
+
+    @Test
+    public void shipEndpointContainsCarrierShipObjects(){
+
+        List<Object> response = template.getForObject("/ships", List.class);
+        Ship carrier = new Ship("Carrier", 5);
+        Assertions.assertEquals(true, response.add(carrier));
```

Comment:
Won't the `response` already contain the Carrier ship? Why is a new Carrier ship being added to the response? 🤔

Reply:
Done! Thank you for the link😊